frontend/pages/SoftwarePage/components/SoftwareFiltersModal/SoftwareFiltersModal.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import Slider from "components/forms/fields/Slider";
+import Checkbox from "components/forms/fields/Checkbox";
+import TooltipWrapper from "components/TooltipWrapper";
+import {
+  findOptionBySeverityRange,
+  ISoftwareVulnFilters,
+  ISoftwareVulnFiltersParams,
+  SEVERITY_DROPDOWN_OPTIONS,
+} from "pages/SoftwarePage/SoftwareTitles/SoftwareTable/helpers";
+
+const baseClass = "software-filters-modal";
+
+interface ISoftwareFiltersModalProps {
+  onExit: () => void;
+  onSubmit: (vulnFilters: ISoftwareVulnFilters) => void;
+  vulnFiltersQueryParams: ISoftwareVulnFiltersParams;
+}
+
+const SoftwareFiltersModal = ({
+  onExit,
+  onSubmit,
+  vulnFiltersQueryParams,
+}: ISoftwareFiltersModalProps) => {
+  const [vulnSoftwareFilterEnabled, setVulnSoftwareFilterEnabled] = useState(
```

Comment:
@jacobshandling 
First off, thanks a mil for such a thorough and thoughtful review. My mind with the 3 features + iteration on designs + oncall has been all over the place.

> Don't we usually hold form data as an object in a single state variable? Not blocking but maybe we update to use that pattern on a future iteration. 

We could, I just checked out manage software automations and we build the form data in the handleSaveautomations function instead of in the state. 

I think this is the only comment I won't have addressed in my follow up PR due to time but I also think it's easy to read so not super pressing to refactor.

Reply:
Agreed